feat: Adding custom NotifyType parameter

abhat-koppert opened the pull request:
Added the parameter for a custom NotifyParameter and a basic relevant test

svaningelgem commented:
Hi, could you resolve the conflicts?

svaningelgem commented:
Could you also adjust the `pyi` file? And maybe there is a way to share information between these 2 as otherwise we're just repeating the same title in both files?

svaningelgem commented on logprise/__init__.py:
```diff
@@ -207,15 +208,23 @@ def accumulate_log(self, message: loguru.Message) -> None:
+        title: str = "Script Notifications",
+        notify_type: str | NotifyType = NotifyType.WARNING,
+        body_format: str | NotifyFormat = NotifyFormat.TEXT,
+
```
I don't know why the automated tests were not run, but maybe let ruff run on the code?
This is what I generally do: `ruff format . && ruff check .`

In any case, I think this empty line should go. But ruff should do that for you.

abhat-koppert replied:
Super weird. I use ruff as my main formatter in VS Code. I guess it missed something. Sorry about that. I have committed the latest change with the fix. Lmk if it is alright.

svaningelgem commented on logprise/__init__.py:
```diff
@@ -207,15 +208,23 @@ def accumulate_log(self, message: loguru.Message) -> None:
+        if self.apprise_obj and self.apprise_obj.notify(
+            title=title, notify_type=notify_type, body=message, body_format=body_format
+        ):
+
```
Remove empty line

svaningelgem commented on tests/test_logprise.py:
```diff
@@ -245,7 +247,53 @@ def test_send_notification_parameters(mocker, noop):
+        (NotifyType.SUCCESS, "markdown"),
+    ],
+)
+def test_notification_parameter_types(mocker, noop, notify_type_param, notify_format_param):
```
I tried to run the tests, but they seem to fail. Can you have a look at them?

abhat-koppert replied:
It seems that I messed something up with the merge conflict resolution. I am not too familiar with GH's merge conflict resolution so I think that was the issue. Should be ok now

svaningelgem commented:
Thanks, can you still adjust the pyi-file? I'll see if I can add a check in the automated procedures to enforce consistency.

svaningelgem commented:
Thanks for your contribution. It has been merged now.
I also added the additional checks into the workflows.

Just let me know if you want to add more things or you prefer me to do a minor release now?

abhat-koppert commented:
A minor release would be nice for now.

 I am also running into an issue where I am attempting to send a message to slack via a Slack hook. However, it does not work when I do not use the `send_notification()` method. I have tried to force the flush interval to be much shorter than the predefined 1 hour. The only way the messages are sent is if I use the method mentioned above. 

I added the Slack hook via the `appriser.add()`. Should I have added it as one of the [apprise recommended config files](https://github.com/caronc/apprise/blob/master/apprise/cli.py) as you have mentioned in your README? Or is there something I am missing?

This is how my logger module is structured in the internal package I am building. Super basic, but ideally should get job done imo. The `logger` object exposed from this module should suffice in sending notifications to Slack when ```logger.error()``` is called, right? Or am I missing something?
```python
import os

from logprise import appriser, logger

appriser.add(
    servers=os.getenv("SLACKHOOK"),
)

appriser.flush_interval = 60

# Export logger and appriser for reuse
__all__ = ["logger", "appriser"]
```
